src/index.js

```diff
@@ -51,19 +54,68 @@ var handlers = {
     'ReadContentAtPositionIntent': readContentAtPosition,
 
     'MoreIntent': function() {
-        // repeat last intent action with increased offSet
-        this.emit(this.event.session.attributes.lastIntent, false);
+        const lastIntent = this.event.session.attributes.lastIntent;
+        switch (lastIntent) {
+            case 'GetHeadlinesIntent':
+            case 'GetLatestReviewsIntent':
+            case 'GetOpinionIntent':
+                // repeat last intent action with increased offSet
+                this.event.session.attributes.lastIntent = 'MoreIntent';
+                this.emit(lastIntent);
+                break;
+
+            default:
+                this.emit(':ask', speech.help.reprompt)
+        }
     },
     'GetOpinionIntent': getOpinion,
 
     'GetLatestReviewsIntent': getLatestReviews,
 
-    //The user has specified a review_type. The reviews intents will prompt for one if missing.
-    'ReviewTypeIntent': function() {
+    /**
+     * This may be a topic or a review_type. 
+     * This is because there is some cross-over between the two, so we need a single handler.
+     * We let the appropriate intent handler decide if the entity is valid.
+     * Note - we also use the name 'entity' in the attributes. This is because the user may not
+     * yet have told us which intent they want, but we need to store the value now.
+     */
+    'EntityIntent': function() {
         const slots = this.event.request.intent.slots;
-        if (slots.review_type && slots.review_type.value) {
-            this.event.session.attributes.reviewType = slots.review_type.value;
-            this.emit(this.event.session.attributes.lastIntent)
+        const getEntity = () => {
+            if (slots.topic && slots.topic.value) return slots.topic.value;
+            if (slots.review_type && slots.review_type.value) return slots.review_type.value;
+            return null;
+        };
+
+        const entity = getEntity();
+        const lastIntent = this.event.session.attributes.lastIntent;
+        this.event.session.attributes.lastIntent = "EntityIntent";  //Signals that attributes contains an entity value
+
+        if (entity.toLowerCase() === "sport" && lastIntent === "Launch") {
+            //Special case - show the sports intro if the skill has just been launched
+            this.emit('GetIntroSportIntent')
+        }
+
+        if (entity !== null) {
```

Comment:
There's some redundancy here, we can rewrite the whole thing like this:
```
        if (entity == null) {
            //This should never happen
            console.log(`Missing entity for EntityIntent: ${this.event.request}`);
            this.emit(':ask', speech.help.reprompt)
        }
        switch (lastIntent) {
            case 'GetLatestReviewsIntent':
            case 'GetIntroReviewsIntent' :
                this.event.session.attributes.reviewType = entity;
                this.emit('GetLatestReviewsIntent');
                break;
            case 'GetIntroNewsIntent':
                this.event.session.attributes.topic = entity;
                this.emit('GetHeadlinesIntent');
                break;
            default:
                // No last intent or unexpected last intent
                this.emit(':ask', speech.help.reprompt)
        }
```
(tested)